src/tests/Common/CLRTest.Jit.targets

```diff
@@ -55,60 +55,22 @@ export RunningIlasmRoundTrip=
 ]]>
       </IlasmRoundTripBashScript>
 
-      <IlasmRoundTripBashScript Condition="'$(CLRTestKind)' == 'BuildAndRun'"><![CDATA[
+      <IlasmRoundTripBashScript Condition="'$(CLRTestKind)' == 'BuildAndRun' And '$(IlasmRoundTripIncompatible)' != 'true'"><![CDATA[
```

Comment:
It is surprising that you would need this since it wasn't needed before.  The script should exit at the beginning if the test is incompatible.  (Leaving the logic in allows for a quick manual script modification in order to run such a test when trying to get it to work rather than rebuilding, though that is a fairly niche scenario.)

Reply:
This was needed to fix a few tests that were marked 'incompatible' but still would run the '.cmd' or '.sh' files with RunningIlasmroundtrip=1. It doesn't hurt to have this.

Reply:
(This comment applies to several places.)

Reply:
It's only new because of how the merged tests are structured. The actual tests here don't cause a problem.

Reply:
What does that mean?  What is the behavior with `IlasmRoundTripIncompatible` that is new with merged tests?  How does the script get to this check if the earlier part of the script exits?

Reply:
I don't remember since it was in the middle of the week. Give me a bit to reproduce what the issue was.

Reply:
Ok, I remember now.

The projects `10w3d_cs_do.csproj`, `Bytemark.csproj`, and `CscBench.csproj` are referenced by `superpmicollect.csproj`.

When they are referenced, the binaries for `10w3d_cs_do.csproj`, `Bytemark.csproj`, and `CscBench.csproj` are included in the output for `superpmicollect` in their own respective folders:
<img width="150" alt="image" src="https://github.com/dotnet/runtime/assets/1278959/633fdc6d-e660-47e6-907f-c6fc61febcbd">

In those folders, they include a `.cmd`/`.sh`:
<img width="659" alt="image" src="https://github.com/dotnet/runtime/assets/1278959/2d88858b-b70f-460c-9fe8-33a19a903023">

Is this necessarily a problem? No, but if we run the `.cmd` with roundtripping, it will not be able to find the corresponding `roundtrip.py` file because it's not emitted in this folder. The `.py` file is only emitted in the original location:
<img width="716" alt="image" src="https://github.com/dotnet/runtime/assets/1278959/0f830521-53e2-4fa5-a73a-e96e21abaafa">

I could certainly try to figure out how to either not emit the `.cmd` in the subfolder location for `superpmicollect` or copy the `roundtrip.py` file over. But in the interest of time, it was easier to just disable these 3 tests.

Reply:
This makes sense, though I think it is about marking those 3 as incompatible.  I don't see how it connects to needing to change the generated script.

It looks like superpmicollect.csproj is just copying every target output (@(_SpmiTest), set by the OutputItemType in the ProjectReference.  An eventual fix might just be to then remove the .cmd/.sh items from that list.

Reply:
I didn't see that, but that seems easy to do.

Reply:
Unfortunately, it's not as simple as I thought. There is no way to determine if the tests are built for ilasmroundtrip. We only know this when we run the actual test.